src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3468,3 +3445,355 @@ void c_typecheck_baset::make_constant_index(exprt &expr)
     throw 0;
   }
 }
+
+optionalt<symbol_exprt> c_typecheck_baset::typecheck_gcc_polymorphic_builtin(
+  const irep_idt &identifier,
+  const exprt::operandst &arguments,
+  const source_locationt &source_location)
+{
+  // the arguments need not be type checked just yet, thus do not make
+  // assumptions that would require type checking
+
+  if(
+    identifier == "__sync_fetch_and_add" ||
```

Comment:
While you're here, consider making these `static irep_idt`s to save hashing the string every time this is called

Reply:
Done.